src/server/api/schema.js

```diff
@@ -622,6 +623,59 @@ const rootMutations = {
       const newCampaign = await campaignInstance.save()
       return editCampaign(newCampaign.id, campaign, loaders)
     },
+    copyCampaign: async (_, { id }, { user, loaders }) => {
+      const campaign = await loaders.campaign.load(id)
+      await accessRequired(user, campaign.organization_id, 'ADMIN')
+
+      const campaignInstance = new Campaign({
+        organization_id: campaign.organization_id,
+        title: 'COPY - ' + campaign.title,
+        description: campaign.description,
+        due_by: campaign.dueBy,
+        is_started: false,
+        is_archived: false
+      })
+      const newCampaign = await campaignInstance.save()
+
+      let interactions = await r.knex('interaction_step')
+        .where({campaign_id: id })
+
+      interactions.forEach((interaction, index) => {
+        const copiedInteraction =
+          new InteractionStep({
+            question: interaction.question,
+            script: interaction.script,
+            answer_option: interaction.answer_option,
+            answer_actions: interaction.answer_actions,
+            is_deleted: interaction.is_deleted,
+            campaign_id: newCampaign.id,
+            parent_interaction_id: interaction.parent_interaction_id
```

Comment:
This PR mostly looks good, except the parent_interaction_id needs to get redone (the old ones point to the old campaign).  One possibility is to re-use [updateInteractionSteps](https://github.com/MoveOnOrg/Spoke/blob/main/src/server/api/schema.js#L361) though that seems to assume an hierarchical structure.

The other way would be to basically re-implement some of its pieces -- basically making an `idMap` forEach interactionStep you save with the key the old id and the value being the new id, and then for all the parent_interaction_ids you'll need to update them based on the idMap.